Src/WitsmlExplorer.Api/Services/WitsmlClientProvider.cs

```diff
@@ -40,29 +40,20 @@ public WitsmlClientProvider(IConfiguration configuration, IHttpContextAccessor h
 
             StringValues? authorizationHeader = httpContextAccessor.HttpContext?.Request.Headers["Authorization"];
             StringValues? targetServerHeader = httpContextAccessor.HttpContext?.Request.Headers[WitsmlServerUrlHeader];
-            StringValues? sourceServerHeader = httpContextAccessor.HttpContext?.Request.Headers[WitsmlSourceServerUrlHeader];
 
-            // Use system creds by role from Bearer token
-            if (authorizationHeader?.Count > 0)
+            if (authorizationHeader?.Count > 0 || targetServerHeader?.Count > 0)
```

Comment:
Should we throw on unsatisfied if?

Reply:
Yes I agree. What kind of exception would be most suitable in this case you think?

Reply:
Created a custom one